class.jetpack.php

```diff
@@ -780,6 +781,12 @@ function() {
 		add_filter( 'jetpack_token_processing_url', array( __CLASS__, 'filter_connect_processing_url' ) );
 		add_filter( 'jetpack_token_redirect_url', array( __CLASS__, 'filter_connect_redirect_url' ) );
 		add_filter( 'jetpack_token_request_body', array( __CLASS__, 'filter_token_request_body' ) );
+
+		// Actions for licensing.
+		Licensing_Manager::instance()->initialize();
+		add_action( 'jetpack_licensing_stored_licenses_request_failed', array( $this, 'log_licensing_request_error' ) );
+		add_action( 'jetpack_licensing_stored_licenses_validations_failed', array( $this, 'log_licensing_validation_errors' ) );
+		add_action( 'load-toplevel_page_jetpack', array( $this, 'surface_licensing_error' ) );
```

Comment:
If the data is meant to be displayed in the Jetpack dashboard, maybe we could fetch it from there directly? There is some info about this here:
https://github.com/Automattic/jetpack/blob/4392a22d643eaf812f42c37739bf675ea3ef7f34/_inc/client/README.md#L16-L26
https://github.com/Automattic/jetpack/blob/4392a22d643eaf812f42c37739bf675ea3ef7f34/docs/rest-api.md

This may mean creating a new core API endpoint in `_inc/lib/class.core-rest-api-endpoints.php` to fetch the data from the option where it is stored first.

You could look at a previous Infinity project like the Setup questionnaire to find an example:
8706a5ecc143e7f0f9e68a5d6f038571bd60fa97
d514f1d9ea86fdfda3b7a23c362574be899c4d6e (in your case the data would only be used to display a notice, so this would not have to be that elaborate).

Reply:
Wouldn't this be a bit overkill for surfacing a single error message to the user and then discarding it?

Reply:
It may be, yes. On the other hand, it may also be the first place folks will check if they need to make changes to that notice.